### PR TITLE
EW-656 provided create group API

### PR DESCRIPTION
**Pull request by Fshmit**
# Description
A REST API was implemented to create a group. Domain driven design was used to implement this new API

## Links to Tickets or other pull requests
Ticket: https://ticketsystem.dbildungscloud.de/browse/EW-656
DDD concept: see doc in server repo.

## Changes
The Auto mapper is no longer used.

## Approval for review
- [ ] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### CODE REVIEW

**Review by kristoff-kiefer — CHANGES_REQUESTED** [review]

File: src/modules/group/domain/gruppe.do.ts
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/domain/gruppe.do.ts:1]: DO is deleted ✅

File: src/modules/group/domain/gruppe.repo.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@nestjs/common';
+import { EntityManager } from '@mikro-orm/postgresql';
+import { DomainError } from '../../../shared/error/domain.error.js';
+import { EntityCouldNotBeCreated } from '../../../shared/error/entity-could-not-be-created.error.js';
+import { GruppeEntity } from '../persistence/gruppe.entity.js';
+@Injectable()
+export class GruppenRepository {
+    public constructor(private readonly em: EntityManager) {}
+
+    public async createGruppe(gruppe: GruppeEntity): Promise<Result<GruppeEntity, DomainError>> {
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/domain/gruppe.repo.ts]: Method is renamed to save and mapping is moved into repo ✅

File: src/modules/group/api/gruppe.controller.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Post()
+    @HttpCode(HttpStatus.CREATED)
+    @ApiCreatedResponse({ description: 'The organisation was successfully created.' })
+    @ApiBadRequestResponse({ description: 'The organisation already exists.' })
+    @ApiUnauthorizedResponse({ description: 'Not authorized to create the organisation.' })
+    @ApiForbiddenResponse({ description: 'Not permitted to create the organisation.' })
+    @ApiInternalServerErrorResponse({ description: 'Internal server error while creating the organisation.' })
+    public async createGroup(@Body() params: CreateGroupBodyParams): Promise<GruppenDo<true> | HttpException> {
+        const gruppe: Gruppe = this.gruppenFactory.createGroup(params);
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/api/gruppe.controller.ts]: group Factory, group aggregate and repo do belong to domain layer. This comment is just about Mapping right? 
or is there another issue I don't see? :)

File: src/modules/group/api/create-group.body.params.spec.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            optionen: referenceParams.optionen,
+            differenzierung: referenceParams.differenzierung,
+            bildungsziele: referenceParams.bildungsziele,
+            jahrgangsstufen: referenceParams.jahrgangsstufen,
+            faecher: referenceParams.faecher,
+            referenzgruppen: referenceParams.referenzgruppen,
+            laufzeit: referenceParams.laufzeit,
+        };
+        const mappedParams: CreateGroupBodyParams = plainToInstance(CreateGroupBodyParams, incomingParams, {});
+        expect(mappedParams).toBeInstanceOf(CreateGroupBodyParams);
```
Comment (psachmann) [inline src/modules/group/api/create-group.body.params.spec.ts:58]: Missing empty line

File: src/modules/group/api/create-group.body.params.spec.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            },
+        ],
+        laufzeit: {
+            von: faker.date.past(),
+            vonLernperiode: '2022',
+            bis: faker.date.future(),
+            bisLernperiode: '2023',
+        },
+    };
+    it('should convert a plain object to a class of createGroupBodyParams', () => {
```
Comment (psachmann) [inline src/modules/group/api/create-group.body.params.spec.ts:41]: Missing empty line

File: src/modules/group/domain/gruppe.spec.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                expect(createdGroup.beschreibung).toEqual(createGroupBodyParams.beschreibung);
+                expect(createdGroup.typ).toEqual(createGroupBodyParams.typ);
+                expect(createdGroup.bereich).toEqual(createGroupBodyParams.bereich);
+                expect(createdGroup.optionen).toEqual(createGroupBodyParams.optionen);
+                expect(createdGroup.differenzierung).toEqual(createGroupBodyParams.differenzierung);
+                expect(createdGroup.bildungsziele).toEqual(createGroupBodyParams.bildungsziele);
+                expect(createdGroup.jahrgangsstufen).toEqual(createGroupBodyParams.jahrgangsstufen);
+                expect(createdGroup.faecher).toEqual(createGroupBodyParams.faecher);
+                expect(createdGroup.referenzgruppen).toEqual(createGroupBodyParams.referenzgruppen);
+                expect(createdGroup.laufzeit).toEqual(createGroupBodyParams.laufzeit);
```
Comment (psachmann) [inline src/modules/group/domain/gruppe.spec.ts:51]: For better maintainability and compiler checks:
```suggestion
expect(result).toEqual(expect.objectContaining<Gruppe<false>>({
    id: expect.any(String),
    createdAt: expect.any(Date),
    updated: expect.any(Date),
    ...
}))
```

File: src/modules/group/domain/gruppe.spec.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                    jahrgangsstufen: [Jahrgangsstufe.JAHRGANGSSTUFE_1, Jahrgangsstufe.JAHRGANGSSTUFE_2],
+                    faecher: [Faecherkanon.DE, Faecherkanon.MA],
+                    referenzgruppen: [
+                        {
+                            id: faker.lorem.word(),
+                            rollen: [Gruppenrollen.LEHR],
+                        },
+                    ],
+                    laufzeit: new Laufzeit({ von: faker.date.recent(), bis: faker.date.recent() }),
+                };
```
Comment (psachmann) [inline src/modules/group/domain/gruppe.spec.ts:78]: Move into a setup method

File: src/modules/group/domain/gruppe.spec.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+                const gruppe: Gruppe<false> = Gruppe.createGroup(createGroupBodyParams);
+                expect(gruppe).toBeInstanceOf(Gruppe);
```
Comment (psachmann) [inline src/modules/group/domain/gruppe.spec.ts:80]: Missing empty lines

File: src/modules/group/domain/gruppe.spec.ts [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+                expect(gruppe.beschreibung).toEqual(createGroupBodyParams.beschreibung);
+                expect(gruppe.typ).toEqual(createGroupBodyParams.typ);
+                expect(gruppe.bereich).toEqual(createGroupBodyParams.bereich);
+                expect(gruppe.optionen).toEqual(createGroupBodyParams.optionen);
+                expect(gruppe.differenzierung).toEqual(createGroupBodyParams.differenzierung);
+                expect(gruppe.bildungsziele).toEqual(createGroupBodyParams.bildungsziele);
+                expect(gruppe.jahrgangsstufen).toEqual(createGroupBodyParams.jahrgangsstufen);
+                expect(gruppe.faecher).toEqual(createGroupBodyParams.faecher);
+                expect(gruppe.referenzgruppen).toEqual(createGroupBodyParams.referenzgruppen);
+                expect(gruppe.laufzeit).toEqual(createGroupBodyParams.laufzeit);
```
Comment (psachmann) [inline src/modules/group/domain/gruppe.spec.ts:99]: Same suggerstion as above

**Review by psachmann — CHANGES_REQUESTED** [review]

File: src/modules/group/api/create-group.body.params.spec.ts (hunk 3 above)
Comment (Fshmit) [reply src/modules/group/api/create-group.body.params.spec.ts:58]: done

File: src/modules/group/api/create-group.body.params.spec.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/api/create-group.body.params.spec.ts:41]: Done

File: src/modules/group/api/create-group.body.params.ts [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    Gruppenbereich,
+    GruppenTyp,
+    Gruppendifferenzierung,
+    Gruppenoption,
+} from '../domain/gruppe.enums.js';
+import { Jahrgangsstufe } from '../../personenkontext/domain/personenkontext.enums.js';
+import { Type } from 'class-transformer';
+import { Referenzgruppen } from '../domain/referenzgruppen.js';
+import { Laufzeit } from '../persistence/laufzeit.js';
+export class CreateGroupBodyParams {
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/api/create-group.body.params.ts:15]: Done

File: src/modules/group/api/create-group.body.params.ts [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+    @IsString()
+    @ApiProperty({ required: false })
+    public readonly beschreibung?: string;
+
+    @IsEnum(GruppenTyp)
+    @ApiProperty({ enum: GruppenTyp, required: true })
+    public readonly typ!: GruppenTyp;
+
+    @IsEnum(Gruppenbereich)
+    @ApiProperty({ enum: Gruppenbereich, required: false })
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/api/create-group.body.params.ts:39]: Done

File: src/modules/group/api/create-group.body.params.ts [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+    public readonly bereich?: Gruppenbereich;
+
+    @IsArray()
+    @IsEnum(Gruppenoption, { each: true })
+    @ApiProperty({ enum: Gruppenoption, required: false, isArray: true })
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/api/create-group.body.params.ts:44]: done

File: src/modules/group/api/create-group.body.params.ts [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+    @IsArray()
+    @ValidateNested()
+    @Type(() => Referenzgruppen)
+    @ApiProperty({ type: Referenzgruppen, required: false, isArray: true })
+    public readonly referenzgruppen?: Referenzgruppen[];
+
+    @ValidateNested()
+    @Type(() => Laufzeit)
+    @ApiProperty({ type: Laufzeit, required: false })
+    public readonly laufzeit!: Laufzeit;
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/api/create-group.body.params.ts:75]: Done

File: src/modules/group/api/gruppe.controller.spec.ts [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+        );
+
+        const params: CreateGroupBodyParams = {
+            bezeichnung: 'test',
+            typ: GruppenTyp.KURS,
+            differenzierung: Gruppendifferenzierung.E,
+            laufzeit: new Laufzeit({ von: new Date(), bis: new Date() }),
+        };
+        describe('when creating a group is successful', () => {
+            it('should return the created group aggeragte', async () => {
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/api/gruppe.controller.spec.ts:80]: done

File: src/modules/group/api/gruppe.controller.ts [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+    ApiInternalServerErrorResponse,
+} from '@nestjs/swagger';
+import { SchulConnexValidationErrorFilter } from '../../../shared/error/schulconnex-validation-error.filter.js';
+import { CreateGroupBodyParams } from './create-group.body.params.js';
+import { GruppenFactory } from '../domain/gruppe.factory.js';
+import { GruppenRepository } from '../domain/gruppe.repo.js';
+import { Gruppe } from '../domain/gruppe.js';
+import { DomainError } from '../../../shared/error/domain.error.js';
+import { SchulConnexErrorMapper } from '../../../shared/error/schul-connex-error.mapper.js';
+@UseFilters(SchulConnexValidationErrorFilter)
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/api/gruppe.controller.ts:18]: done

File: src/modules/group/api/gruppe.controller.ts [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+    public async createGroup(@Body() params: CreateGroupBodyParams): Promise<Gruppe<true> | HttpException> {
+        const gruppe: Gruppe<false> = this.gruppenFactory.createGroup(params);
+        const result: Result<Gruppe<true>, DomainError> = await this.gruppenRepository.save(gruppe);
+        if (!result.ok) {
+            return SchulConnexErrorMapper.mapSchulConnexErrorToHttpException(
+                SchulConnexErrorMapper.mapDomainErrorToSchulConnexError(result.error),
+            );
+        }
+        return result.value;
+    }
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/api/gruppe.controller.ts:44]: done

File: src/modules/group/domain/gruppe.mapper.ts [hunk 16]
```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@nestjs/common';
+import { GruppeEntity } from '../persistence/gruppe.entity.js';
+import { Gruppe } from './gruppe.js';
+@Injectable()
+export class GruppeMapper {
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/domain/gruppe.mapper.ts:5]: done

File: src/modules/group/domain/gruppe.mapper.ts [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+        gruppeEntity.bereich = gruppe.bereich;
+        gruppeEntity.optionen = gruppe.optionen;
+        gruppeEntity.differenzierung = gruppe.differenzierung;
+        gruppeEntity.bildungsziele = gruppe.bildungsziele;
+        gruppeEntity.jahrgangsstufen = gruppe.jahrgangsstufen;
+        gruppeEntity.faecher = gruppe.faecher;
+        gruppeEntity.referenzgruppen = gruppe.referenzgruppen;
+        gruppeEntity.laufzeit = gruppe.laufzeit;
+        return gruppeEntity;
+    }
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/domain/gruppe.mapper.ts:24]: done

File: src/modules/group/domain/gruppe.repo.spec.ts [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+    });
+
+    afterAll(async () => {
+        await module.close();
+    });
+
+    beforeEach(() => {
+        jest.resetAllMocks();
+    });
+    describe('when creating a group with invalid data', () => {
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/domain/gruppe.repo.spec.ts:47]: done

File: src/modules/group/domain/gruppe.repo.ts [hunk 19]
```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@nestjs/common';
+import { EntityManager } from '@mikro-orm/postgresql';
+import { DomainError } from '../../../shared/error/domain.error.js';
+import { EntityCouldNotBeCreated } from '../../../shared/error/entity-could-not-be-created.error.js';
+import { GruppeEntity } from '../persistence/gruppe.entity.js';
+import { Gruppe } from './gruppe.js';
+import { GruppeMapper } from './gruppe.mapper.js';
+@Injectable()
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/domain/gruppe.repo.ts:8]: done

File: src/modules/group/domain/gruppe.spec.ts [hunk 20]
```diff
@@ -0,0 +1,103 @@
...
+    GruppenTyp,
+    Gruppenbereich,
+    Gruppendifferenzierung,
+    Gruppenoption,
+    Gruppenrollen,
+} from './gruppe.enums.js';
+import { CreateGroupBodyParams } from '../api/create-group.body.params.js';
+import { Jahrgangsstufe } from '../../personenkontext/domain/personenkontext.enums.js';
+import { Laufzeit } from '../persistence/laufzeit.js';
+describe('Gruppe', () => {
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/domain/gruppe.spec.ts:15]: done

File: src/modules/group/domain/gruppe.ts [hunk 21]
```diff
@@ -0,0 +1,117 @@
...
+import {
+    Bildungsziele,
+    Faecherkanon,
+    Gruppenbereich,
+    GruppenTyp,
+    Gruppendifferenzierung,
+    Gruppenoption,
+} from './gruppe.enums.js';
+import { Laufzeit } from '../persistence/laufzeit.js';
+export class Gruppe<WasPersisted extends boolean> {
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/domain/gruppe.ts:13]: done

File: src/modules/group/persistence/gruppe.entity.ts [hunk 22]
```diff
@@ -0,0 +1,71 @@
...
+import { Jahrgangsstufe, SichtfreigabeType } from '../../personenkontext/domain/personenkontext.enums.js';
+import { Referenzgruppen } from '../domain/referenzgruppen.js';
+import { Laufzeit } from './laufzeit.js';
+
+@Entity({ tableName: 'gruppe' })
+export class GruppeEntity extends TimestampedEntity {
+    public constructor() {
+        super();
+    }
+
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/persistence/gruppe.entity.ts:20]: done

File: src/modules/group/persistence/gruppe.entity.ts [hunk 23]
```diff
@@ -0,0 +1,71 @@
...
+    @Enum({ items: () => Faecherkanon, nullable: true, array: true })
+    public faecher?: Faecherkanon[];
+
+    @Embedded(() => Referenzgruppen, { nullable: true, array: true })
+    public referenzgruppen?: Referenzgruppen[];
+
+    @Embedded(() => Laufzeit, { nullable: false })
+    public laufzeit?: Laufzeit;
+
+    @Property({ nullable: false, default: SichtfreigabeType.NEIN })
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/persistence/gruppe.entity.ts:66]: no revision is not an enum

File: src/modules/group/persistence/laufzeit.ts [hunk 24]
```diff
@@ -0,0 +1,39 @@
+import { Embeddable, Property } from '@mikro-orm/core';
+import { ApiProperty } from '@nestjs/swagger';
+import { Type } from 'class-transformer';
+import { IsDate, IsOptional, IsString } from 'class-validator';
+
+@Embeddable()
+export class Laufzeit {
+    @IsDate()
+    @IsOptional()
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/persistence/laufzeit.ts:9]: removed

File: src/modules/group/api/create-group.body.params.spec.ts [hunk 25]
```diff
@@ -0,0 +1,61 @@
+import { faker } from '@faker-js/faker';
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/api/create-group.body.params.spec.ts:1]: Yes, the test was required, but is no longer needed. The test is therefore deleted ✅

File: src/modules/group/api/create-group.body.params.ts [hunk 26]
```diff
@@ -0,0 +1,76 @@
...
+    Gruppendifferenzierung,
+    Gruppenoption,
+} from '../domain/gruppe.enums.js';
+import { Jahrgangsstufe } from '../../personenkontext/domain/personenkontext.enums.js';
+import { Type } from 'class-transformer';
+import { Referenzgruppen } from '../domain/referenzgruppen.js';
+import { Laufzeit } from '../persistence/laufzeit.js';
+export class CreateGroupBodyParams {
+    @IsString()
+    @ApiProperty({ required: false })
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/api/create-group.body.params.ts:19]: Done ✅

File: src/modules/group/api/gruppe.controller.spec.ts [hunk 27]
```diff
@@ -0,0 +1,104 @@
...
+import { CreateGroupBodyParams } from './create-group.body.params.js';
+import { GruppenTyp, Gruppenbereich, Gruppendifferenzierung, Gruppenoption } from '../domain/gruppe.enums.js';
+import { Gruppe } from '../domain/gruppe.js';
+import { Laufzeit } from '../persistence/laufzeit.js';
+import { GruppenRepository } from '../domain/gruppe.repo.js';
+import { HttpException } from '@nestjs/common';
+import { GruppeMapper } from '../domain/gruppe.mapper.js';
+import { EntityCouldNotBeCreated } from '../../../shared/error/entity-could-not-be-created.error.js';
+import { faker } from '@faker-js/faker';
+describe('GruppeController', () => {
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/api/gruppe.controller.spec.ts]: Done ✅

File: src/modules/group/api/gruppe.controller.spec.ts [hunk 28]
```diff
@@ -0,0 +1,104 @@
...
+        repo = module.get(GruppenRepository);
+    });
+
+    afterAll(async () => {
+        await module.close();
+    });
+
+    beforeAll(() => {
+        jest.resetAllMocks();
+    });
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/api/gruppe.controller.spec.ts]: Si 👍

File: src/console/dbseed/db-seed.service.ts [hunk 29]
```diff
@@ -22,6 +23,19 @@ export class DbSeedService {
...
+
+    public readLernperiode(fileContentAsStr: string): LernperiodeFile[] {
+        const entities: LernperiodeFile[] = this.readEntityFromJSONFile<LernperiodeFile>(
+            fileContentAsStr,
+            () => new LernperiodeFile(),
+        );
+        for (const entity of entities) {
+            this.lernperiodeMap.set(entity.id, entity);
+        }
+        return entities;
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/console/dbseed/db-seed.service.ts:37]: Done ✅

File: src/modules/group/domain/gruppe.mapper.spec.ts [hunk 30]
```diff
@@ -0,0 +1,125 @@
...
+    Gruppenoption,
+    Gruppendifferenzierung,
+    Bildungsziele,
+    Faecherkanon,
+    Gruppenrollen,
+} from './gruppe.enums.js';
+import { GruppeMapper } from './gruppe.mapper.js';
+import { Referenzgruppen } from './referenzgruppen.js';
+// AI next 107 lines
+describe('GruppeMapper', () => {
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/domain/gruppe.mapper.spec.ts]: Done ✅

File: src/modules/group/domain/gruppe.spec.ts (hunk 7 above)
Comment (Fshmit) [reply src/modules/group/domain/gruppe.spec.ts:81]: Done ✅

File: src/modules/group/domain/gruppe.repo.integration-spec.ts [hunk 31]
```diff
@@ -0,0 +1,91 @@
...
+
+    describe('createGruppe', () => {
+        describe('when creating a gruppe', () => {
+            it('should create gruppe', async () => {
+                const result: Result<Gruppe<true>, DomainError> = await repo.save(gruppe);
+
+                expect(result).toBeDefined();
+                await expect(em.find(GruppeEntity, {})).resolves.toHaveLength(1);
+                expect(result.ok).toBe(true);
+            });
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/domain/gruppe.repo.integration-spec.ts:89]: Done

File: src/modules/group/domain/gruppe.spec.ts (hunk 5 above)
Comment (Fshmit) [reply src/modules/group/domain/gruppe.spec.ts:52]: linter will throw an error:   
    id: expect.any(String),
    createdAt: expect.any(Date),
    updated: expect.any(Date),
a possible solution will look like this: 

```suggestion

                expect(createdGroup).toEqual(
                    expect.objectContaining({
                        id: undefined,
                        createdAt: undefined,
                        updatedAt: undefined,
                        revision: '',
                        mandant: '',
                        organisationId: '',
                        referrer: createGroupBodyParams.referrer,
                        thema: createGroupBodyParams.thema,
                        bereich: createGroupBodyParams.bereich,
                        beschreibung: createGroupBodyParams.beschreibung,
                        bezeichnung: createGroupBodyParams.bezeichnung,
                        typ: createGroupBodyParams.typ,
                        optionen: createGroupBodyParams.optionen,
                        differenzierung: createGroupBodyParams.differenzierung,
                        bildungsziele: createGroupBodyParams.bildungsziele,
                        jahrgangsstufen: createGroupBodyParams.jahrgangsstufen,
```

File: src/modules/group/domain/gruppe.factory.spec.ts [hunk 32]
```diff
@@ -0,0 +1,81 @@
...
+        [],
+    );
+
+    beforeAll(() => {
+        factory = new GruppenFactory();
+    });
+
+    describe('when creating a group', () => {
+        it('should create a group aggregate', () => {
+            jest.spyOn(Gruppe, 'createGroup').mockReturnValue(gruppe as unknown as Gruppe<false>);
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/domain/gruppe.factory.spec.ts:64]: Done

File: src/modules/group/domain/gruppe.mapper.spec.ts [hunk 33]
```diff
@@ -0,0 +1,125 @@
...
+            expect(result.thema).toBe(gruppeEntity.thema);
+            expect(result.beschreibung).toBe(gruppeEntity.beschreibung);
+            expect(result.bereich).toBe(gruppeEntity.bereich);
+            expect(result.optionen).toBe(gruppeEntity.optionen);
+            expect(result.differenzierung).toBe(gruppeEntity.differenzierung);
+            expect(result.bildungsziele).toBe(gruppeEntity.bildungsziele);
+            expect(result.jahrgangsstufen).toBe(gruppeEntity.jahrgangsstufen);
+            expect(result.faecher).toBe(gruppeEntity.faecher);
+            expect(result.referenzgruppen).toBe(gruppeEntity.referenzgruppen);
+            expect(result.laufzeit).toBe(gruppeEntity.laufzeit);
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/domain/gruppe.mapper.spec.ts:123]: Done

File: src/modules/group/domain/gruppe.mapper.spec.ts [hunk 34]
```diff
@@ -0,0 +1,125 @@
...
+            expect(result.beschreibung).toBe(gruppe.beschreibung);
+            expect(result.typ).toBe(gruppe.typ);
+            expect(result.bereich).toBe(gruppe.bereich);
+            expect(result.optionen).toBe(gruppe.optionen);
+            expect(result.differenzierung).toBe(gruppe.differenzierung);
+            expect(result.bildungsziele).toBe(gruppe.bildungsziele);
+            expect(result.jahrgangsstufen).toBe(gruppe.jahrgangsstufen);
+            expect(result.faecher).toBe(gruppe.faecher);
+            expect(result.referenzgruppen).toBe(gruppe.referenzgruppen);
+            expect(result.laufzeit).toBe(gruppe.laufzeit);
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/domain/gruppe.mapper.spec.ts:74]: Done

File: src/modules/group/gruppe.module.ts [hunk 35]
```diff
@@ -0,0 +1,13 @@
+import { Module } from '@nestjs/common';
```
[the remark this answers is not in the document]
Comment (Fshmit) [reply src/modules/group/gruppe.module.ts:1]: Done ✅

**Comment by pkleybolte** [comment]
Outdated. Partly done already.

**Pull request closed without merging**